src/utilities/BlockTime.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.26;
+
+import "openzeppelin-contracts/contracts/access/AccessControl.sol";
+import "openzeppelin-contracts/contracts/utils/ReentrancyGuard.sol";
+import "src/utilities/TimeToken.sol";
+import "src/interfaces/IBlockTime.sol";
+
+contract BlockTime is IBlockTime, AccessControl, ReentrancyGuard {
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant TIME_KEEPER = keccak256("TIME_KEEPER");
+    bytes32 public constant SCHEDULER_ROLE = keccak256("SCHEDULER_ROLE");
+
+    /// @dev minimum number of signed time values needed
+    uint256 public minNumberOfChronicles;
+
+    /// @notice value used to ensure the values being provided are not outliers
+    uint256 public maxBlockWidth;
+
+    /// @notice since the precision of time is low in this implementation we assume time to be anywhere
+    ///         between current mean time and current mean time + time width
+    ///         This can be in the future be modified to be an average of the difference between last X earthTimeValues
+    uint256 public timeBlockWidth;
+
+    /// @notice the current average of all time keepers provided time value
+    uint256 public currentEarthTimeAvg;
+
+    /// @notice The ERC20 timeToken that will be transferred to users on successfull time updation
+    TimeToken public timeToken; 
+
+    event Tick(uint256 currentEarthTimeBlockStart, uint256 currentEarthTimeBlockEnd);
+    event EarthTimeUpdated(uint256 newEarthTime, Chronicle[] chronicles);
+    event MaxBlockWidthSet(uint256 maxBlockWidth);
+
+    constructor(string memory _name, string memory _symbol) {
+        timeToken = new TimeToken(_name, _symbol, address(this));
+        _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        _grantRole(ADMIN_ROLE, _msgSender());
+    }
+
+    /// @notice changes earth avg time
+    function moveTime(Chronicle[] calldata chronicles, uint256 meanCurrentEarthTime, address[] calldata receivers, uint256[] calldata amounts) external onlyRole(SCHEDULER_ROLE) nonReentrant {
+        currentEarthTimeAvg = meanCurrentEarthTime;
+        timeToken.batchMint(receivers, amounts);
+        emit EarthTimeUpdated(meanCurrentEarthTime, chronicles);
```

Comment:
Should also emit receivers and amounts

Reply:
It is being emitted in the TimeToken contract

Reply:
Please add it here as well